pkg/webui/lib/components/date-time/index.js

```diff
@@ -0,0 +1,127 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { FormattedDate, FormattedTime } from 'react-intl'
+import bind from 'autobind-decorator'
+
+import PropTypes from '../../prop-types'
+import RelativeTime from './relative'
+
+@bind
+class DateTime extends React.PureComponent {
+
+  renderDateTime (formattedDate, formattedTime, dateValue) {
+    const {
+      className,
+      children,
+      date,
+      time,
+    } = this.props
+    const t = []
+
+    const formattedDateTime = dateValue.toISOString()
+
+    if (date) {
+      t.push(formattedDate)
+    }
+
+    if (time) {
+      if (date) {
+        t.push(' ')
+      }
+
+      t.push(formattedTime)
+    }
+
+    const result = t.join('')
```

Comment:
Progressively concatenating a string using `+=` would seem easier to me to achieve this.